lte/gateway/c/oai/tasks/sgw/spgw_state_converter.cpp

```diff
@@ -104,14 +104,16 @@ void SpgwStateConverter::spgw_bearer_context_to_proto(
   sgw_eps_bearer_proto->set_mme_teid_s11(sgw_eps_bearer_state->mme_teid_S11);
   bstring ip_addr_bstr =
       ip_address_to_bstring(&sgw_eps_bearer_state->mme_ip_address_S11);
-  sgw_eps_bearer_proto->set_mme_ip_address_s11(bdata(ip_addr_bstr));
+  BSTRING_TO_STRING(
```

Comment:
Is there a better way for us to enforce this type safety?

Reply:
The main difference between these two changes is that the macro takes care of writing taking into account the length of the bstring and using that to parse, we could add tests (unit testing is something that we're very much missing for all the conversions) that verifies the size / length of the parsed and unparsed data.